### ISSUE TITLE
txlist API timeout for addresses with large number of transactions

### ISSUE THREAD

**Issue by alexeystepanyan** [poanetwork/blockscout#1319]
*Describe your issue here.*

When using blockscout API for ETC blockchain txlist times out for accounts that have large number of transactions

Example request:

https://blockscout.com/etc/mainnet/api?module=account&action=txlist&address=0xa6E1B726eF41E7D18Df6858E9F4eD76012fC2c2E&startblock=0&endblock=9999999999&sort=desc 

requests.exceptions.HTTPError: 504 Server Error: Gateway Time-out

I have tried using pagination with page 1 and offset 10 

https://blockscout.com/etc/mainnet/api?module=account&action=txlist&address=0xa6E1B726eF41E7D18Df6858E9F4eD76012fC2c2E&startblock=0&endblock=9999999999&sort=desc&page=1&offset=10

but still have the same result with timeout

Removing sorting and block params and leaving pagination for only 10 transactions still times out.

https://blockscout.com/etc/mainnet/api?module=account&action=txlist&address=0xa6E1B726eF41E7D18Df6858E9F4eD76012fC2c2E&page=1&offset=10

requests.exceptions.HTTPError: 504 Server Error: Gateway Time-out

Could you help diagnose this? blockscout UI shows transactions for this address and loads them lazily, could API support this as well?

Thanks!

**Comment by zachdaniel** [comment]
@vbaranov Looked into this a bit more, and I checked an address on mainnet with over 10k transactions, and it loads fine. Do we perhaps just need to beef up the ethereum classic hardware? I can look for an optimization as well, just want to check with you first.

https://blockscout.com/eth/mainnet/api?module=account&action=txlist&address=0xe71efa3ac6895450da7c154576849d5091a86627&startblock=0&endblock=9999999999&sort=desc

**Comment by vbaranov** [comment]
@zachdaniel 

I cannot make it work even with `offset=1` for ETC:

https://blockscout.com/etc/mainnet/api?module=account&action=txlist&address=0xa6E1B726eF41E7D18Df6858E9F4eD76012fC2c2E&page=1&offset=1

**Comment by zachdaniel** [comment]
Yeah, but the performance issue has to do with the amount of transactions in the account vs the amount you are requesting. The main point here is that requesting a much larger transaction list on mainnet is fine, leading me to believe it's related to the more and not the code.

**Comment by febarajan** [comment]
But things are not working....

**Comment by febarajan** [comment]
Failed to Add Address transactions { FetchError: invalid json response body at https://blockscout.com/etc/mainnet/api?module=account&action=txlist&address=0xd144e30a0571aaf0d0c050070ac435deba461fab
